packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/rest_api/FlutterRestApi.kt

```diff
@@ -0,0 +1,269 @@
+                                prepareError(
+                                        flutterResult,
+                                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                                        cancelToken,
+                                        error)
+                            }
+                    )
+                } else {
+                    operation = functionWithApiName(
+                            apiName,
+                            options,
+                            Consumer { result ->
+                                if (!cancelToken.isNullOrEmpty()) OperationsManager.removeOperation(cancelToken)
+                                prepareRestResponseResult(flutterResult, result, methodName) },
+                            Consumer { error ->
+                                if (!cancelToken.isNullOrEmpty()) OperationsManager.removeOperation(cancelToken)
+                                prepareError(
+                                        flutterResult,
+                                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                                        cancelToken,
+                                        error)
+                            }
+                    )
+                }
+                OperationsManager.addOperation(cancelToken, operation!!)
+
+            } catch (e: Exception) {
+                prepareError(
+                        flutterResult,
+                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                        cancelToken,
+                        e)
+            }
+        }
+
+        fun prepareError(flutterResult: Result, msg: String, cancelToken: String, error: Exception) {
+            FlutterApiError.postFlutterError(flutterResult, msg, error)
+        }
+
+        private fun prepareRestResponseResult(flutterResult: Result, result: RestResponse, methodName: String) {
+
+            var restResponse = FlutterSerializedRestResponse(result)
+
+            // if code is not 200 then throw an exception
+            /*
+            result.code.toString = "Code{" +
+                "statusCode=" + statusCode +
+                '}';
+             */
+            if (!result.code.isSuccessful) {
+                FlutterApiError.handleAPIError(
+                        flutterResult,
+                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                        "The HTTP response status code is [" + result.code.toString().substring(16, 19) + "].",
+                        """
+                    The metadata associated with the response is contained in the HTTPURLResponse.
+                    For more information on HTTP status codes, take a look at
+                    https://en.wikipedia.org/wiki/List_of_HTTP_status_codes
```

Comment:
Can we move this to a constant so that this method call is more readable and less verbose?